libbeat/docs/monitoring/configuring.asciidoc

```diff
@@ -0,0 +1,178 @@
+//// descriptions here generic enough to work for all Beats that include
+//// this file. When using cross references, make sure that the cross
+//// references resolve correctly for any files that include this one.
+//// Use the appropriate variables defined in the index.asciidoc file to
+//// resolve Beat names: beatname_uc and beatname_lc.
+//// Use the following include to pull this content into a doc file:
+//// include::../../libbeat/docs/monitoring/configuring.asciidoc[]
+//// Make sure this content appears below a level 2 heading.
+//////////////////////////////////////////////////////////////////////////
+
+[role="xpack"]
+[[configuring-monitoring]]
+== Configuring monitoring for {beatname_uc}
+++++
+<titleabbrev>Configure monitoring</titleabbrev>
+++++
+
+{monitoring} enables you to easily monitor {beatname_uc} from {kib}. For more
+information, see {kibana-ref}/xpack-monitoring.html[Monitoring].
+
+. Configure {beatname_uc} to send metrics to your {es} cluster by setting the
+`xpack.monitoring` settings in the {beatname_uc} configuration file. For
+example:
++
+--
+[source, yml]
+--------------------
+xpack.monitoring:
+  enabled: true
+  elasticsearch:
+    url: ["https://example.com:9200", "https://example2.com:9200"]
+    username: elastic
+    password: changeme
+--------------------
+
+Many of the configuration options for `xpack.monitoring` match the
+`output.elasticsearch` options. If you configured {es} output and you want to
+use the same {es} cluster and credentials to store your monitoring metrics, you
+can specify the following minimal configuration options:
+
+[source, yml]
+--------------------
+xpack.monitoring:
+  enabled: true
+  elasticsearch:
+--------------------
+
+--
+
+. {kibana-ref}/monitoring-xpack-kibana.html[Configure monitoring in {kib}].
+
+. To verify your monitoring configuration, point your web browser at your {kib}
+host, and select Monitoring from the side navigation. Metrics reported from
+{beatname_uc} should be visible in the Beats section. When {security} is enabled,
+to view the monitoring dashboards you must log in to {kib} as a user who has the
+`kibana_user` and `monitoring_user` roles.
+
+[float]
+=== Configuration options
+
+You can specify the following options in the `xpack.monitoring` section of the
++{beatname_lc}.yml+ config file:
+
+[float]
+==== `enabled`
+
+The `enabled` config is a boolean setting to enable or disable {monitoring}.
+If set to `true`, monitoring is enabled.
+
+The default value is `false`.
+
+[float]
+==== `elasticsearch`
+
+The {es} instances that you want to ship your {beatname_uc} metrics to. This
+configuration option contains the following fields:
+
+[float]
+===== `bulk_max_size`
+
+The maximum number of metrics to bulk in a single {es} bulk API index request.
+The default is `50`. For more information, see <<elasticsearch-output>>.
+
+////
+[float]
+===== `buffer_size`
+
+TBD. The default value is `50`.
+////
```

Comment:
This setting is obsolete. It's not used anymore. Will remove it from code base in follow up PR.